src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,321 @@
+        }
+        catch (IOException ex) {
+            throw new CheckstyleException("Failed to write javadoc properties of '" + inputFile
+                + "' to '" + outputFile + "'", ex);
+        }
+        finally {
+            CommonUtils.close(writer);
+        }
+    }
+
+    /**
+     * Checks that the DetailAST is a {@code public} {@code static} {@code final} {@code int} field.
+     * @param ast to process
+     * @return {@code true} if matches; {@code false} otherwise
+     */
+    private static boolean isPublicStaticFinalIntField(DetailAST ast) {
+        boolean result = ast.getType() == TokenTypes.VARIABLE_DEF;
+        if (result) {
+            final DetailAST type = ast.findFirstToken(TokenTypes.TYPE);
+            result = type.getFirstChild().getType() == TokenTypes.LITERAL_INT;
+            if (result) {
+                final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
+                result = modifiers.findFirstToken(TokenTypes.LITERAL_PUBLIC) != null
+                    && modifiers.findFirstToken(TokenTypes.LITERAL_STATIC) != null
+                    && modifiers.findFirstToken(TokenTypes.FINAL) != null;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Extracts the name of an ast.
+     * @param ast to extract the name
+     * @return the text content of the inner {@code TokenTypes.IDENT} node
+     */
+    private static String getName(DetailAST ast) {
+        return ast.findFirstToken(TokenTypes.IDENT).getText();
+    }
+
+    /**
+     * Extracts the first sentence of the comment of an DetailAST.
+     * @param ast to extract the first sentence
+     * @return the first sentence of the inner {@code TokenTypes.BLOCK_COMMENT_BEGIN} node
+     *      or {@code null} if the first sentence is absent or malformed (does not end with period)
+     * @throws CheckstyleException if a javadoc comment can not be parsed or an unsupported inline
+     *      tag found
+     */
+    private static String getFirstJavadocSentence(DetailAST ast) throws CheckstyleException {
+        String firstSentence = null;
+        for (DetailAST child = ast.getFirstChild(); child != null && firstSentence == null;
+                child = child.getNextSibling()) {
+            // If there is an annotation, the javadoc comment will be a child of it.
+            if (child.getType() == TokenTypes.ANNOTATION) {
+                firstSentence = getFirstJavadocSentence(child);
+            }
+            // Otherwise, the javadoc comment will be right here.
+            else if (child.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
+                    && JavadocUtils.isJavadocComment(child)) {
+                final DetailNode tree = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(child);
+                firstSentence = getFirstJavadocSentence(tree);
+            }
+        }
+        return firstSentence;
+    }
+
+    /**
+     * Extracts the first sentence of the comment of an DetailAST.
```

Comment:
please make a note that we cannot use logic from SummaryJavadocCheck as here we need conversion from javadoc inline tags to HTML.

Reply:
Perhaps this code should be moved to somewhere else. JavadocUtils or something like that.
Then both the SummaryJavadocCheck and this generator can use it.

Reply:
done